gateway-ha/src/test/java/io/trino/gateway/ha/TestTrinoResource.java

```diff
@@ -0,0 +1,273 @@
+        assertEquals(5678, groups[2].getResourceGroupId());
+    }
+
+    @Test
+    @Order(2)
+    public void testReadResourceGroupsByGroupId()
+            throws Exception
+    {
+        Request request =
+                new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/trino/resourcegroup/read/1234")
+                        .get()
+                        .build();
+        Response response = httpClient.newCall(request).execute();
+        assertEquals(200, response.code());
+
+        ResourceGroupsDetail[] resourceGroups =
+                objectMapper.readValue(response.body().string(), ResourceGroupsDetail[].class);
+        assertEquals(1, resourceGroups.length);
+
+        assertEquals(1234, resourceGroups[0].getResourceGroupId());
+    }
+
+    @Test
+    @Order(3)
+    public void testReadSelectorsAll()
+            throws Exception
+    {
+        Request request =
+                new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/trino/selector/read/")
+                        .get()
+                        .build();
+        Response response = httpClient.newCall(request).execute();
+        assertEquals(200, response.code());
+
+        SelectorsDetail[] selectors =
+                objectMapper.readValue(response.body().string(), SelectorsDetail[].class);
+        assertEquals(2, selectors.length);
+
+        Arrays.sort(selectors, (x, y) -> Long.compare(x.getResourceGroupId(), y.getResourceGroupId()));
+        assertEquals(1234, selectors[0].getResourceGroupId());
+        assertEquals(3456, selectors[1].getResourceGroupId());
+    }
+
+    @Test
+    @Order(4)
+    public void testReadSelectorsByGroupId()
+            throws Exception
+    {
+        Request request =
+                new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/trino/selector/read/3456")
+                        .get()
+                        .build();
+        Response response = httpClient.newCall(request).execute();
+        assertEquals(200, response.code());
+
+        SelectorsDetail[] selectors =
+                objectMapper.readValue(response.body().string(), SelectorsDetail[].class);
+        assertEquals(1, selectors.length);
+
+        assertEquals(3456, selectors[0].getResourceGroupId());
+    }
+
+    @Test
+    @Order(5)
+    public void testDeleteResourceGroupOk()
+            throws Exception
+    {
+        RequestBody requestBody =
+                RequestBody.create(MediaType.parse("application/json; charset=utf-8"), "");
+        Request request =
+                new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/trino/resourcegroup/delete/5678")
+                        .post(requestBody)
+                        .build();
+        Response response = httpClient.newCall(request).execute();
+        assertEquals(200, response.code());
```

Comment:
Use HttpStatus.OK_200 from jetty .. and similar ones in tests below

Reply:
Done